src/main/java/com/twilio/http/TwilioRestClient.java

```diff
@@ -72,22 +88,32 @@ public Response request(final Request request) {
 
         logRequest(request);
         Response response = httpClient.reliableRequest(request);
+        if(response != null) {
+            int statusCode = response.getStatusCode();
+            if (statusCode == HTTP_STATUS_CODE_UNAUTHORIZED && authStrategy != null && EnumConstants.AuthType.TOKEN.equals(authStrategy.getAuthType())) {
+                ((TokenAuthStrategy)authStrategy).fetchToken();
```

Comment:
Moving fetchtoken logic inside auth strategy class and just fetching the auth string would be more aligned to the concept of abstraction

Reply:
Discussed over call, this code is required for fetching token in case there is **401**
Will be adding fetchToken in getAuthString() method as mentioned in previous comment.